l2geth/core/vm/evm.go

```diff
@@ -393,6 +398,20 @@ func (evm *EVM) create(caller ContractRef, codeAndHash *codeAndHash, gas uint64,
 	if !evm.CanTransfer(evm.StateDB, caller.Address(), value) {
 		return nil, common.Address{}, gas, ErrInsufficientBalance
 	}
+	if rcfg.UsingOVM {
+		// Make sure the creator address should be able to deploy.
+		if !evm.AddressWhitelisted(caller.Address()) {
+			// Try to encode this error as a Solidity error message so it's more clear to end-users
+			// what's going on when a contract creation fails.
+			solerr := fmt.Errorf("deployer address not whitelisted: %s", caller.Address().Hex())
```

Comment:
In the tests, is it possible to say `.to.be.reverted.with("deployer address...` to cover this functionality?